t/0700-unifyfs-stage-full.t

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# Test unifyfs-stage executable for basic functionality
+#
+
+test_description="Test basic functionality of unifyfs-stage executable"
+
+. $(dirname $0)/sharness.sh
+
+test_expect_success "unifyfs-stage exists" '
+    test_path_is_file ${SHARNESS_BUILD_DIRECTORY}/util/unifyfs-stage/src/unifyfs-stage
+'
+test_expect_success "testing temp dir exists" '
+    test_path_is_dir  ${UNIFYFS_TEST_TMPDIR}
+'
+
+mkdir -p ${UNIFYFS_TEST_TMPDIR}/config_0700
+mkdir -p ${UNIFYFS_TEST_TMPDIR}/stage_source
+mkdir -p ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700
+
+test_expect_success "stage testing dirs exist" '
+    test_path_is_dir  ${UNIFYFS_TEST_TMPDIR}/config_0700
+    test_path_is_dir  ${UNIFYFS_TEST_TMPDIR}/stage_source
+    test_path_is_dir  ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700
+'
+
+dd if=/dev/urandom bs=4M count=1 of=${UNIFYFS_TEST_TMPDIR}/stage_source/source_0700.file
+
+test_expect_success "source.file exists" '
+    test_path_is_file ${UNIFYFS_TEST_TMPDIR}/stage_source/source_0700.file
+'
+
+rm -f ${UNIFYFS_TEST_TMPDIR}/config_0700/*
+rm -f ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700/*
+
+test_expect_success "config_0700 directory is empty" '
+    test_dir_is_empty ${UNIFYFS_TEST_TMPDIR}/config_0700
+'
+
+echo "\"${UNIFYFS_TEST_TMPDIR}/stage_source/source_0700.file\" \"${UNIFYFS_TEST_MOUNT}/intermediate.file\""  > ${UNIFYFS_TEST_TMPDIR}/config_0700/test_IN.manifest
+echo "\"${UNIFYFS_TEST_MOUNT}/intermediate.file\" \"${UNIFYFS_TEST_TMPDIR}/stage_destination_0700/destination_0700.file\"" > ${UNIFYFS_TEST_TMPDIR}/config_0700/test_OUT.manifest
+
+test_expect_success "config_0700 directory now has manifest files" '
+    test_path_is_file  ${UNIFYFS_TEST_TMPDIR}/config_0700/test_IN.manifest
+    test_path_is_file  ${UNIFYFS_TEST_TMPDIR}/config_0700/test_OUT.manifest
+'
+
+test_expect_success "target directory is empty" '
+    test_dir_is_empty ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700
+'
+
+${SHARNESS_BUILD_DIRECTORY}/util/unifyfs-stage/src/unifyfs-stage -m ${UNIFYFS_TEST_MOUNT} ${UNIFYFS_TEST_TMPDIR}/config_0700/test_IN.manifest > ${UNIFYFS_TEST_TMPDIR}/config_0700/stage_IN_output.OUT 2>&1
+
+${SHARNESS_BUILD_DIRECTORY}/util/unifyfs-stage/src/unifyfs-stage -m ${UNIFYFS_TEST_MOUNT} ${UNIFYFS_TEST_TMPDIR}/config_0700/test_OUT.manifest > ${UNIFYFS_TEST_TMPDIR}/config_0700/stage_OUT_output.OUT 2>&1
+
+test_expect_success "input file has been staged to output" '
+    test_path_is_file ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700/destination_0700.file
+'
+
+test_expect_failure "final output is identical to initial input" '
+    test_cmp ${UNIFYFS_TEST_TMPDIR}/stage_source/source_0700.file ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700/destination_0700.file
+'
```

Comment:
Hmm. Well. I realize I suggested this, but now that I think about it, let's make one more change to this since you have an issue to track it:
```suggestion
test_expect_failure "final output is identical to initial input" '
    test_might_fail test_cmp ${UNIFYFS_TEST_TMPDIR}/stage_source/source_0700.file ${UNIFYFS_TEST_TMPDIR}/stage_destination_0700/destination_0700.file
'
```
I say this because as `test_expect_failure` will make this test actually fail when run locally and through GitLab CI.
Could always comment it out or just remove it and eventually add it to integration test suite, but just adding the `test_might_fail` before it will still let it run but essentially label it as "unstable."

Reply:
I was a bit squeamish about that myself, but I didn't know about the test_might_fail directive.  I completely agree.  

Checked uploaded and checked in, and Travis is happy with it.

Reply:
Nice! I'd forgotten it was a thing as I hadn't used it before.